examples/jacobi.cxx

```diff
@@ -0,0 +1,333 @@
+// IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include <cstdlib>
+#include <cstdio> 
+#include <iostream>
+
+#include "RAJA/RAJA.hpp"
+#include "RAJA/util/defines.hpp"
+
+
+#define CUDA_BLOCK_SIZE 16
+
+template <typename T>
+T* allocate(RAJA::Index_type size) {
+  T* ptr;
+#if defined(RAJA_ENABLE_CUDA)
+  cudaMallocManaged((void**)&ptr, sizeof(T)*size,cudaMemAttachGlobal);
+#else
+  ptr = new T[size];
+#endif
+  return ptr;
+}
+
+template <typename T>
+void deallocate(T* &ptr) {
+  if (ptr) {
+#if defined(RAJA_ENABLE_CUDA)
+    cudaFree(ptr);
+#else
+    delete[] ptr;
+#endif
+    ptr = nullptr;
+  }
+}
+
+
+
+/*Example 3: Jacobi Method For Solving a Linear System
+
+  ----[Details]--------------------
+  The following code uses the jacobi method to compute the currents in a
+  structured grid by solving a linear system correspoding to kirchoff's 
+  circuit law. 
+
+  The structured N x N circuit may be visualized as follows: 
+
+  - Dash lines correspond to lossless wires,
+  - The circles correspond to lossless juntions
+  - The [] correspond to 1 Ohm resistors
+  - The [x] corresponds to a 1 volt DC battery which powers the grid
+  
+  o--[]--o--[]--o--[]--o--[]--o
+  |      |      |      |      |  
+  []    []     []     []     []     
+  |      |      |      |      |
+  o--[]--o--[]--o--[]--o--[]--o
+  |      |      |      |      |  
+  []    []     []     []     []     
+  |      |      |      |      |
+  o--[]--o--[]--o--[]--o--[]--o
+  |      |      |      |      |  
+  []    []     []     []     []     
+  |      |      |      |      |
+  o--[]--o--[]--o--[]--o--[]--o
+  |      |      |      |      |  
+  [x]    []     []     []     []     
+  |      |      |      |      |
+  o--[]--o--[]--o--[]--o--[]--o
+
+
+  ----[RAJA Concepts]---------------
+  1. RAJA Reduction
+  2. RAJA::omp_collapse_nowait_exec
+
+*/
+
+int main(int RAJA_UNUSED_ARG(argc), char** RAJA_UNUSED_ARG(argv[]))
+{
+
+  std::cout<<"Example 3: Jacobi Method for Linear System"<<std::endl;
+
+  //----[Parameters for Solver]-------
+  double tol = 1e-5;
+  int maxIter = 10000;
+  int N = 100;
+  int NN=(N+2)*(N+2);
+  int iteration;
+  double resI2, V, invD;
+
+  double *I    = allocate<double>(NN);
+  double *Iold = allocate<double>(NN);
+
+  //Set to zero
+  memset(I,0,NN*sizeof(double));
+  memset(Iold,0,NN*sizeof(double));
+
+
+  //----[Standard C++ Loop]------
+  printf("Standard  C++ Loop \n");
+  resI2 = 1;  iteration = 0; 
+
+  //Carry out iterations of the Jacobi scheme until a tolerance is met
+  while(resI2>tol*tol) {
+    
+    //Iteration of the Jacobi Scheme
+    for(int n=1;n<=N;++n) {
+      for(int m=1;m<=N;++m) {
+
+        int id = n*(N+2) + m;
+
+        //Cell (1,1) is a special case due to battery
+        if(n==1 && m==1) {
+          invD = 1./3.0; V = 1.0; 
```

Comment:
for the sake of a user looking at these examples, it would be nice to know what invD and V are. You mention the parameters but do not describe them.